// initapi/src/type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.Data = exports.ApiInfo = exports.EnumInfo = exports.EntityInfo = exports.ReferenceObject = exports.isSchemaObjectTypeArray = exports.isReferenceObject = void 0;
/** 判读是否 ReferenceObject 类型 */
function isReferenceObject(obj) {
    return obj.hasOwnProperty('$ref');
}
exports.isReferenceObject = isReferenceObject;
/** 判断是否 SchemaObjectType[] 类型 */
const isSchemaObjectTypeArray = (type) => {
    return Array.isArray(type);
};
exports.isSchemaObjectTypeArray = isSchemaObjectTypeArray;
class ReferenceObject {
}
exports.ReferenceObject = ReferenceObject;
class EntityInfo {
    constructor() {
        /** entity名 */
        this.name = '';
        /** 备注 */
        this.desc = '';
        /** 属性信息 */
        this.propList = [];
    }
}
exports.EntityInfo = EntityInfo;
class EnumInfo {
    constructor() {
        /** enum名 */
        this.name = '';
        /** 备注 */
        this.desc = '';
        /** 枚举信息 */
        this.enumList = [];
    }
}
exports.EnumInfo = EnumInfo;
class ApiInfo {
    constructor() {
        /** 后端的接口方法名 */
        this.funcName = '';
        /** api地址 */
        this.url = '';
        /** 概要 */
        this.summary = '';
        /** 备注 */
        this.desc = '';
        /** 请求方式 */
        this.mode = '';
        /** 路由传参 */
        this.path = [];
        /** ?传参 */
        this.params = [];
        /** 响应体 */
        this.res = {};
    }
}
exports.ApiInfo = ApiInfo;
class Data {
    constructor() {
        /** data属性列表 */
        this.propList = [];
        // propList?: (SchemaObject | ReferenceObject)[] = []
    }
}
exports.Data = Data;
/** path, params 的类型放在 schema 属性里面 */
class Params {
    constructor() {
        // $ref?: string
        /** 参数名 */
        this.name = '';
        /** 备注 */
        this.desc = '';
        // TODO 与语义不符
        /** 是否必填(true 必填, false 可选) */
        this.nullable = false;
        /** 参数类型 */
        // schema?: SchemaObject = undefined
        this.schema = {};
    }
}
exports.Params = Params;
